core/integration/testdata/modules/go/ifaces/main.go

```diff
@@ -0,0 +1,24 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/exec"
+)
+
+type Caller struct{}
+
+func (m *Caller) Test(
+	ctx context.Context,
+	// +optional
+	run string,
+) error {
+	args := []string{"test", "-v", "-count=1", "."}
+	if run != "" {
+		args = append(args, "-run", run)
+	}
+	cmd := exec.CommandContext(ctx, "go", args...)
```

Comment:
Yeah the thought process here was that I needed to write a lot of tests, the tests for interfaces inherently required multiple dependencies be loaded at the same time, and that trying to write all of it as raw graphql queries sounded like a nightmare.

Which ended up bringing me to this pattern. I kind of like it? The progress output ends up working out nicely since it's all a nested session (due to how modules work) so you just see essentially the same `go test` output seen when running any of the other tests. 

It also hints at the idea that's been brought up previously of having first-class support for writing module test code; I guess you can think of this as doing "by-hand" what that first-class support would do in the background.

Reply:
Neat - thought it might be something like that!